// @aws-sdk/client-route-53/dist-es/commands/ActivateKeySigningKeyCommand.js
import { __extends } from "tslib";
import { getIdNormalizerPlugin } from "@aws-sdk/middleware-sdk-route53";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ActivateKeySigningKeyRequestFilterSensitiveLog, ActivateKeySigningKeyResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restXmlActivateKeySigningKeyCommand, serializeAws_restXmlActivateKeySigningKeyCommand, } from "../protocols/Aws_restXml";
var ActivateKeySigningKeyCommand = (function (_super) {
    __extends(ActivateKeySigningKeyCommand, _super);
    function ActivateKeySigningKeyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ActivateKeySigningKeyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getIdNormalizerPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "Route53Client";
        var commandName = "ActivateKeySigningKeyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ActivateKeySigningKeyRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ActivateKeySigningKeyResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ActivateKeySigningKeyCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlActivateKeySigningKeyCommand(input, context);
    };
    ActivateKeySigningKeyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlActivateKeySigningKeyCommand(output, context);
    };
    return ActivateKeySigningKeyCommand;
}($Command));
export { ActivateKeySigningKeyCommand };
