// @aws-sdk/client-route-53/dist-es/commands/AssociateVPCWithHostedZoneCommand.js
import { __extends } from "tslib";
import { getIdNormalizerPlugin } from "@aws-sdk/middleware-sdk-route53";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { AssociateVPCWithHostedZoneRequestFilterSensitiveLog, AssociateVPCWithHostedZoneResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restXmlAssociateVPCWithHostedZoneCommand, serializeAws_restXmlAssociateVPCWithHostedZoneCommand, } from "../protocols/Aws_restXml";
var AssociateVPCWithHostedZoneCommand = (function (_super) {
    __extends(AssociateVPCWithHostedZoneCommand, _super);
    function AssociateVPCWithHostedZoneCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    AssociateVPCWithHostedZoneCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getIdNormalizerPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "Route53Client";
        var commandName = "AssociateVPCWithHostedZoneCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: AssociateVPCWithHostedZoneRequestFilterSensitiveLog,
            outputFilterSensitiveLog: AssociateVPCWithHostedZoneResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    AssociateVPCWithHostedZoneCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlAssociateVPCWithHostedZoneCommand(input, context);
    };
    AssociateVPCWithHostedZoneCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlAssociateVPCWithHostedZoneCommand(output, context);
    };
    return AssociateVPCWithHostedZoneCommand;
}($Command));
export { AssociateVPCWithHostedZoneCommand };
