// @aws-sdk/client-route-53/dist-es/commands/CreateCidrCollectionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateCidrCollectionRequestFilterSensitiveLog, CreateCidrCollectionResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restXmlCreateCidrCollectionCommand, serializeAws_restXmlCreateCidrCollectionCommand, } from "../protocols/Aws_restXml";
var CreateCidrCollectionCommand = (function (_super) {
    __extends(CreateCidrCollectionCommand, _super);
    function CreateCidrCollectionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateCidrCollectionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "Route53Client";
        var commandName = "CreateCidrCollectionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateCidrCollectionRequestFilterSensitiveLog,
            outputFilterSensitiveLog: CreateCidrCollectionResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateCidrCollectionCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlCreateCidrCollectionCommand(input, context);
    };
    CreateCidrCollectionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlCreateCidrCollectionCommand(output, context);
    };
    return CreateCidrCollectionCommand;
}($Command));
export { CreateCidrCollectionCommand };
