// @aws-sdk/client-route-53/dist-es/commands/CreateHostedZoneCommand.js
import { __extends } from "tslib";
import { getIdNormalizerPlugin } from "@aws-sdk/middleware-sdk-route53";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateHostedZoneRequestFilterSensitiveLog, CreateHostedZoneResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restXmlCreateHostedZoneCommand, serializeAws_restXmlCreateHostedZoneCommand, } from "../protocols/Aws_restXml";
var CreateHostedZoneCommand = (function (_super) {
    __extends(CreateHostedZoneCommand, _super);
    function CreateHostedZoneCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateHostedZoneCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getIdNormalizerPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "Route53Client";
        var commandName = "CreateHostedZoneCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateHostedZoneRequestFilterSensitiveLog,
            outputFilterSensitiveLog: CreateHostedZoneResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateHostedZoneCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlCreateHostedZoneCommand(input, context);
    };
    CreateHostedZoneCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlCreateHostedZoneCommand(output, context);
    };
    return CreateHostedZoneCommand;
}($Command));
export { CreateHostedZoneCommand };
