// @aws-sdk/client-route-53/dist-es/commands/CreateHealthCheckCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateHealthCheckRequestFilterSensitiveLog, CreateHealthCheckResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restXmlCreateHealthCheckCommand, serializeAws_restXmlCreateHealthCheckCommand, } from "../protocols/Aws_restXml";
var CreateHealthCheckCommand = (function (_super) {
    __extends(CreateHealthCheckCommand, _super);
    function CreateHealthCheckCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateHealthCheckCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "Route53Client";
        var commandName = "CreateHealthCheckCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateHealthCheckRequestFilterSensitiveLog,
            outputFilterSensitiveLog: CreateHealthCheckResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateHealthCheckCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlCreateHealthCheckCommand(input, context);
    };
    CreateHealthCheckCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlCreateHealthCheckCommand(output, context);
    };
    return CreateHealthCheckCommand;
}($Command));
export { CreateHealthCheckCommand };
