// @aws-sdk/client-route-53/dist-es/commands/CreateReusableDelegationSetCommand.js
import { __extends } from "tslib";
import { getIdNormalizerPlugin } from "@aws-sdk/middleware-sdk-route53";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateReusableDelegationSetRequestFilterSensitiveLog, CreateReusableDelegationSetResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restXmlCreateReusableDelegationSetCommand, serializeAws_restXmlCreateReusableDelegationSetCommand, } from "../protocols/Aws_restXml";
var CreateReusableDelegationSetCommand = (function (_super) {
    __extends(CreateReusableDelegationSetCommand, _super);
    function CreateReusableDelegationSetCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateReusableDelegationSetCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getIdNormalizerPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "Route53Client";
        var commandName = "CreateReusableDelegationSetCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateReusableDelegationSetRequestFilterSensitiveLog,
            outputFilterSensitiveLog: CreateReusableDelegationSetResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateReusableDelegationSetCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlCreateReusableDelegationSetCommand(input, context);
    };
    CreateReusableDelegationSetCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlCreateReusableDelegationSetCommand(output, context);
    };
    return CreateReusableDelegationSetCommand;
}($Command));
export { CreateReusableDelegationSetCommand };
