// @aws-sdk/client-route-53/dist-es/commands/CreateTrafficPolicyVersionCommand.js
import { __extends } from "tslib";
import { getIdNormalizerPlugin } from "@aws-sdk/middleware-sdk-route53";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateTrafficPolicyVersionRequestFilterSensitiveLog, CreateTrafficPolicyVersionResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restXmlCreateTrafficPolicyVersionCommand, serializeAws_restXmlCreateTrafficPolicyVersionCommand, } from "../protocols/Aws_restXml";
var CreateTrafficPolicyVersionCommand = (function (_super) {
    __extends(CreateTrafficPolicyVersionCommand, _super);
    function CreateTrafficPolicyVersionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateTrafficPolicyVersionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getIdNormalizerPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "Route53Client";
        var commandName = "CreateTrafficPolicyVersionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateTrafficPolicyVersionRequestFilterSensitiveLog,
            outputFilterSensitiveLog: CreateTrafficPolicyVersionResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateTrafficPolicyVersionCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlCreateTrafficPolicyVersionCommand(input, context);
    };
    CreateTrafficPolicyVersionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlCreateTrafficPolicyVersionCommand(output, context);
    };
    return CreateTrafficPolicyVersionCommand;
}($Command));
export { CreateTrafficPolicyVersionCommand };
