// @aws-sdk/client-route-53/dist-es/commands/DeleteQueryLoggingConfigCommand.js
import { __extends } from "tslib";
import { getIdNormalizerPlugin } from "@aws-sdk/middleware-sdk-route53";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteQueryLoggingConfigRequestFilterSensitiveLog, DeleteQueryLoggingConfigResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restXmlDeleteQueryLoggingConfigCommand, serializeAws_restXmlDeleteQueryLoggingConfigCommand, } from "../protocols/Aws_restXml";
var DeleteQueryLoggingConfigCommand = (function (_super) {
    __extends(DeleteQueryLoggingConfigCommand, _super);
    function DeleteQueryLoggingConfigCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteQueryLoggingConfigCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getIdNormalizerPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "Route53Client";
        var commandName = "DeleteQueryLoggingConfigCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteQueryLoggingConfigRequestFilterSensitiveLog,
            outputFilterSensitiveLog: DeleteQueryLoggingConfigResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteQueryLoggingConfigCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlDeleteQueryLoggingConfigCommand(input, context);
    };
    DeleteQueryLoggingConfigCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlDeleteQueryLoggingConfigCommand(output, context);
    };
    return DeleteQueryLoggingConfigCommand;
}($Command));
export { DeleteQueryLoggingConfigCommand };
