// @aws-sdk/client-route-53/dist-es/commands/DeleteHealthCheckCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteHealthCheckRequestFilterSensitiveLog, DeleteHealthCheckResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restXmlDeleteHealthCheckCommand, serializeAws_restXmlDeleteHealthCheckCommand, } from "../protocols/Aws_restXml";
var DeleteHealthCheckCommand = (function (_super) {
    __extends(DeleteHealthCheckCommand, _super);
    function DeleteHealthCheckCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteHealthCheckCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "Route53Client";
        var commandName = "DeleteHealthCheckCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteHealthCheckRequestFilterSensitiveLog,
            outputFilterSensitiveLog: DeleteHealthCheckResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteHealthCheckCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlDeleteHealthCheckCommand(input, context);
    };
    DeleteHealthCheckCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlDeleteHealthCheckCommand(output, context);
    };
    return DeleteHealthCheckCommand;
}($Command));
export { DeleteHealthCheckCommand };
