// @aws-sdk/client-route-53/dist-es/commands/DeleteHostedZoneCommand.js
import { __extends } from "tslib";
import { getIdNormalizerPlugin } from "@aws-sdk/middleware-sdk-route53";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteHostedZoneRequestFilterSensitiveLog, DeleteHostedZoneResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restXmlDeleteHostedZoneCommand, serializeAws_restXmlDeleteHostedZoneCommand, } from "../protocols/Aws_restXml";
var DeleteHostedZoneCommand = (function (_super) {
    __extends(DeleteHostedZoneCommand, _super);
    function DeleteHostedZoneCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteHostedZoneCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getIdNormalizerPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "Route53Client";
        var commandName = "DeleteHostedZoneCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteHostedZoneRequestFilterSensitiveLog,
            outputFilterSensitiveLog: DeleteHostedZoneResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteHostedZoneCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlDeleteHostedZoneCommand(input, context);
    };
    DeleteHostedZoneCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlDeleteHostedZoneCommand(output, context);
    };
    return DeleteHostedZoneCommand;
}($Command));
export { DeleteHostedZoneCommand };
