// @aws-sdk/client-route-53/dist-es/commands/DeleteTrafficPolicyInstanceCommand.js
import { __extends } from "tslib";
import { getIdNormalizerPlugin } from "@aws-sdk/middleware-sdk-route53";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteTrafficPolicyInstanceRequestFilterSensitiveLog, DeleteTrafficPolicyInstanceResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restXmlDeleteTrafficPolicyInstanceCommand, serializeAws_restXmlDeleteTrafficPolicyInstanceCommand, } from "../protocols/Aws_restXml";
var DeleteTrafficPolicyInstanceCommand = (function (_super) {
    __extends(DeleteTrafficPolicyInstanceCommand, _super);
    function DeleteTrafficPolicyInstanceCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteTrafficPolicyInstanceCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getIdNormalizerPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "Route53Client";
        var commandName = "DeleteTrafficPolicyInstanceCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteTrafficPolicyInstanceRequestFilterSensitiveLog,
            outputFilterSensitiveLog: DeleteTrafficPolicyInstanceResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteTrafficPolicyInstanceCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlDeleteTrafficPolicyInstanceCommand(input, context);
    };
    DeleteTrafficPolicyInstanceCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlDeleteTrafficPolicyInstanceCommand(output, context);
    };
    return DeleteTrafficPolicyInstanceCommand;
}($Command));
export { DeleteTrafficPolicyInstanceCommand };
