// @aws-sdk/client-route-53/dist-es/commands/GetAccountLimitCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetAccountLimitRequestFilterSensitiveLog, GetAccountLimitResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restXmlGetAccountLimitCommand, serializeAws_restXmlGetAccountLimitCommand, } from "../protocols/Aws_restXml";
var GetAccountLimitCommand = (function (_super) {
    __extends(GetAccountLimitCommand, _super);
    function GetAccountLimitCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetAccountLimitCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "Route53Client";
        var commandName = "GetAccountLimitCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetAccountLimitRequestFilterSensitiveLog,
            outputFilterSensitiveLog: GetAccountLimitResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetAccountLimitCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlGetAccountLimitCommand(input, context);
    };
    GetAccountLimitCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlGetAccountLimitCommand(output, context);
    };
    return GetAccountLimitCommand;
}($Command));
export { GetAccountLimitCommand };
