// @aws-sdk/client-route-53/dist-es/commands/GetCheckerIpRangesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetCheckerIpRangesRequestFilterSensitiveLog, GetCheckerIpRangesResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restXmlGetCheckerIpRangesCommand, serializeAws_restXmlGetCheckerIpRangesCommand, } from "../protocols/Aws_restXml";
var GetCheckerIpRangesCommand = (function (_super) {
    __extends(GetCheckerIpRangesCommand, _super);
    function GetCheckerIpRangesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetCheckerIpRangesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "Route53Client";
        var commandName = "GetCheckerIpRangesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetCheckerIpRangesRequestFilterSensitiveLog,
            outputFilterSensitiveLog: GetCheckerIpRangesResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetCheckerIpRangesCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlGetCheckerIpRangesCommand(input, context);
    };
    GetCheckerIpRangesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlGetCheckerIpRangesCommand(output, context);
    };
    return GetCheckerIpRangesCommand;
}($Command));
export { GetCheckerIpRangesCommand };
