// @aws-sdk/client-route-53/dist-es/commands/GetChangeCommand.js
import { __extends } from "tslib";
import { getIdNormalizerPlugin } from "@aws-sdk/middleware-sdk-route53";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetChangeRequestFilterSensitiveLog, GetChangeResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restXmlGetChangeCommand, serializeAws_restXmlGetChangeCommand } from "../protocols/Aws_restXml";
var GetChangeCommand = (function (_super) {
    __extends(GetChangeCommand, _super);
    function GetChangeCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetChangeCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getIdNormalizerPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "Route53Client";
        var commandName = "GetChangeCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetChangeRequestFilterSensitiveLog,
            outputFilterSensitiveLog: GetChangeResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetChangeCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlGetChangeCommand(input, context);
    };
    GetChangeCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlGetChangeCommand(output, context);
    };
    return GetChangeCommand;
}($Command));
export { GetChangeCommand };
