// @aws-sdk/client-route-53/dist-es/commands/GetDNSSECCommand.js
import { __extends } from "tslib";
import { getIdNormalizerPlugin } from "@aws-sdk/middleware-sdk-route53";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetDNSSECRequestFilterSensitiveLog, GetDNSSECResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restXmlGetDNSSECCommand, serializeAws_restXmlGetDNSSECCommand } from "../protocols/Aws_restXml";
var GetDNSSECCommand = (function (_super) {
    __extends(GetDNSSECCommand, _super);
    function GetDNSSECCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetDNSSECCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getIdNormalizerPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "Route53Client";
        var commandName = "GetDNSSECCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetDNSSECRequestFilterSensitiveLog,
            outputFilterSensitiveLog: GetDNSSECResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetDNSSECCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlGetDNSSECCommand(input, context);
    };
    GetDNSSECCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlGetDNSSECCommand(output, context);
    };
    return GetDNSSECCommand;
}($Command));
export { GetDNSSECCommand };
