// @aws-sdk/client-route-53/dist-es/commands/GetHostedZoneCommand.js
import { __extends } from "tslib";
import { getIdNormalizerPlugin } from "@aws-sdk/middleware-sdk-route53";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetHostedZoneRequestFilterSensitiveLog, GetHostedZoneResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restXmlGetHostedZoneCommand, serializeAws_restXmlGetHostedZoneCommand, } from "../protocols/Aws_restXml";
var GetHostedZoneCommand = (function (_super) {
    __extends(GetHostedZoneCommand, _super);
    function GetHostedZoneCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetHostedZoneCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getIdNormalizerPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "Route53Client";
        var commandName = "GetHostedZoneCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetHostedZoneRequestFilterSensitiveLog,
            outputFilterSensitiveLog: GetHostedZoneResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetHostedZoneCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlGetHostedZoneCommand(input, context);
    };
    GetHostedZoneCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlGetHostedZoneCommand(output, context);
    };
    return GetHostedZoneCommand;
}($Command));
export { GetHostedZoneCommand };
