// @aws-sdk/client-route-53/dist-es/commands/GetHealthCheckCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetHealthCheckRequestFilterSensitiveLog, GetHealthCheckResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restXmlGetHealthCheckCommand, serializeAws_restXmlGetHealthCheckCommand, } from "../protocols/Aws_restXml";
var GetHealthCheckCommand = (function (_super) {
    __extends(GetHealthCheckCommand, _super);
    function GetHealthCheckCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetHealthCheckCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "Route53Client";
        var commandName = "GetHealthCheckCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetHealthCheckRequestFilterSensitiveLog,
            outputFilterSensitiveLog: GetHealthCheckResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetHealthCheckCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlGetHealthCheckCommand(input, context);
    };
    GetHealthCheckCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlGetHealthCheckCommand(output, context);
    };
    return GetHealthCheckCommand;
}($Command));
export { GetHealthCheckCommand };
