// @aws-sdk/client-route-53/dist-es/commands/GetHealthCheckLastFailureReasonCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetHealthCheckLastFailureReasonRequestFilterSensitiveLog, GetHealthCheckLastFailureReasonResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restXmlGetHealthCheckLastFailureReasonCommand, serializeAws_restXmlGetHealthCheckLastFailureReasonCommand, } from "../protocols/Aws_restXml";
var GetHealthCheckLastFailureReasonCommand = (function (_super) {
    __extends(GetHealthCheckLastFailureReasonCommand, _super);
    function GetHealthCheckLastFailureReasonCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetHealthCheckLastFailureReasonCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "Route53Client";
        var commandName = "GetHealthCheckLastFailureReasonCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetHealthCheckLastFailureReasonRequestFilterSensitiveLog,
            outputFilterSensitiveLog: GetHealthCheckLastFailureReasonResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetHealthCheckLastFailureReasonCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlGetHealthCheckLastFailureReasonCommand(input, context);
    };
    GetHealthCheckLastFailureReasonCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlGetHealthCheckLastFailureReasonCommand(output, context);
    };
    return GetHealthCheckLastFailureReasonCommand;
}($Command));
export { GetHealthCheckLastFailureReasonCommand };
