// @aws-sdk/client-route-53/dist-es/commands/GetHostedZoneCountCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetHostedZoneCountRequestFilterSensitiveLog, GetHostedZoneCountResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restXmlGetHostedZoneCountCommand, serializeAws_restXmlGetHostedZoneCountCommand, } from "../protocols/Aws_restXml";
var GetHostedZoneCountCommand = (function (_super) {
    __extends(GetHostedZoneCountCommand, _super);
    function GetHostedZoneCountCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetHostedZoneCountCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "Route53Client";
        var commandName = "GetHostedZoneCountCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetHostedZoneCountRequestFilterSensitiveLog,
            outputFilterSensitiveLog: GetHostedZoneCountResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetHostedZoneCountCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlGetHostedZoneCountCommand(input, context);
    };
    GetHostedZoneCountCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlGetHostedZoneCountCommand(output, context);
    };
    return GetHostedZoneCountCommand;
}($Command));
export { GetHostedZoneCountCommand };
