// @aws-sdk/client-route-53/dist-es/commands/GetReusableDelegationSetLimitCommand.js
import { __extends } from "tslib";
import { getIdNormalizerPlugin } from "@aws-sdk/middleware-sdk-route53";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetReusableDelegationSetLimitRequestFilterSensitiveLog, GetReusableDelegationSetLimitResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restXmlGetReusableDelegationSetLimitCommand, serializeAws_restXmlGetReusableDelegationSetLimitCommand, } from "../protocols/Aws_restXml";
var GetReusableDelegationSetLimitCommand = (function (_super) {
    __extends(GetReusableDelegationSetLimitCommand, _super);
    function GetReusableDelegationSetLimitCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetReusableDelegationSetLimitCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getIdNormalizerPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "Route53Client";
        var commandName = "GetReusableDelegationSetLimitCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetReusableDelegationSetLimitRequestFilterSensitiveLog,
            outputFilterSensitiveLog: GetReusableDelegationSetLimitResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetReusableDelegationSetLimitCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlGetReusableDelegationSetLimitCommand(input, context);
    };
    GetReusableDelegationSetLimitCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlGetReusableDelegationSetLimitCommand(output, context);
    };
    return GetReusableDelegationSetLimitCommand;
}($Command));
export { GetReusableDelegationSetLimitCommand };
