// @aws-sdk/client-route-53/dist-es/commands/ListTrafficPolicyVersionsCommand.js
import { __extends } from "tslib";
import { getIdNormalizerPlugin } from "@aws-sdk/middleware-sdk-route53";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListTrafficPolicyVersionsRequestFilterSensitiveLog, ListTrafficPolicyVersionsResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restXmlListTrafficPolicyVersionsCommand, serializeAws_restXmlListTrafficPolicyVersionsCommand, } from "../protocols/Aws_restXml";
var ListTrafficPolicyVersionsCommand = (function (_super) {
    __extends(ListTrafficPolicyVersionsCommand, _super);
    function ListTrafficPolicyVersionsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListTrafficPolicyVersionsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getIdNormalizerPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "Route53Client";
        var commandName = "ListTrafficPolicyVersionsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListTrafficPolicyVersionsRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ListTrafficPolicyVersionsResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListTrafficPolicyVersionsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlListTrafficPolicyVersionsCommand(input, context);
    };
    ListTrafficPolicyVersionsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlListTrafficPolicyVersionsCommand(output, context);
    };
    return ListTrafficPolicyVersionsCommand;
}($Command));
export { ListTrafficPolicyVersionsCommand };
