// @aws-sdk/client-route-53/dist-es/commands/ListGeoLocationsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListGeoLocationsRequestFilterSensitiveLog, ListGeoLocationsResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restXmlListGeoLocationsCommand, serializeAws_restXmlListGeoLocationsCommand, } from "../protocols/Aws_restXml";
var ListGeoLocationsCommand = (function (_super) {
    __extends(ListGeoLocationsCommand, _super);
    function ListGeoLocationsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListGeoLocationsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "Route53Client";
        var commandName = "ListGeoLocationsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListGeoLocationsRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ListGeoLocationsResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListGeoLocationsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlListGeoLocationsCommand(input, context);
    };
    ListGeoLocationsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlListGeoLocationsCommand(output, context);
    };
    return ListGeoLocationsCommand;
}($Command));
export { ListGeoLocationsCommand };
