// @aws-sdk/client-route-53/dist-es/commands/ListResourceRecordSetsCommand.js
import { __extends } from "tslib";
import { getIdNormalizerPlugin } from "@aws-sdk/middleware-sdk-route53";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListResourceRecordSetsRequestFilterSensitiveLog, ListResourceRecordSetsResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restXmlListResourceRecordSetsCommand, serializeAws_restXmlListResourceRecordSetsCommand, } from "../protocols/Aws_restXml";
var ListResourceRecordSetsCommand = (function (_super) {
    __extends(ListResourceRecordSetsCommand, _super);
    function ListResourceRecordSetsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListResourceRecordSetsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getIdNormalizerPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "Route53Client";
        var commandName = "ListResourceRecordSetsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListResourceRecordSetsRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ListResourceRecordSetsResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListResourceRecordSetsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlListResourceRecordSetsCommand(input, context);
    };
    ListResourceRecordSetsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlListResourceRecordSetsCommand(output, context);
    };
    return ListResourceRecordSetsCommand;
}($Command));
export { ListResourceRecordSetsCommand };
