// @aws-sdk/client-route-53/dist-es/commands/ListCidrBlocksCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListCidrBlocksRequestFilterSensitiveLog, ListCidrBlocksResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restXmlListCidrBlocksCommand, serializeAws_restXmlListCidrBlocksCommand, } from "../protocols/Aws_restXml";
var ListCidrBlocksCommand = (function (_super) {
    __extends(ListCidrBlocksCommand, _super);
    function ListCidrBlocksCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListCidrBlocksCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "Route53Client";
        var commandName = "ListCidrBlocksCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListCidrBlocksRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ListCidrBlocksResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListCidrBlocksCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlListCidrBlocksCommand(input, context);
    };
    ListCidrBlocksCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlListCidrBlocksCommand(output, context);
    };
    return ListCidrBlocksCommand;
}($Command));
export { ListCidrBlocksCommand };
