// @aws-sdk/client-route-53/dist-es/commands/ListCidrCollectionsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListCidrCollectionsRequestFilterSensitiveLog, ListCidrCollectionsResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restXmlListCidrCollectionsCommand, serializeAws_restXmlListCidrCollectionsCommand, } from "../protocols/Aws_restXml";
var ListCidrCollectionsCommand = (function (_super) {
    __extends(ListCidrCollectionsCommand, _super);
    function ListCidrCollectionsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListCidrCollectionsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "Route53Client";
        var commandName = "ListCidrCollectionsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListCidrCollectionsRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ListCidrCollectionsResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListCidrCollectionsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlListCidrCollectionsCommand(input, context);
    };
    ListCidrCollectionsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlListCidrCollectionsCommand(output, context);
    };
    return ListCidrCollectionsCommand;
}($Command));
export { ListCidrCollectionsCommand };
