// @aws-sdk/client-route-53/dist-es/commands/ListTrafficPolicyInstancesByPolicyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListTrafficPolicyInstancesByPolicyRequestFilterSensitiveLog, ListTrafficPolicyInstancesByPolicyResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restXmlListTrafficPolicyInstancesByPolicyCommand, serializeAws_restXmlListTrafficPolicyInstancesByPolicyCommand, } from "../protocols/Aws_restXml";
var ListTrafficPolicyInstancesByPolicyCommand = (function (_super) {
    __extends(ListTrafficPolicyInstancesByPolicyCommand, _super);
    function ListTrafficPolicyInstancesByPolicyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListTrafficPolicyInstancesByPolicyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "Route53Client";
        var commandName = "ListTrafficPolicyInstancesByPolicyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListTrafficPolicyInstancesByPolicyRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ListTrafficPolicyInstancesByPolicyResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListTrafficPolicyInstancesByPolicyCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlListTrafficPolicyInstancesByPolicyCommand(input, context);
    };
    ListTrafficPolicyInstancesByPolicyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlListTrafficPolicyInstancesByPolicyCommand(output, context);
    };
    return ListTrafficPolicyInstancesByPolicyCommand;
}($Command));
export { ListTrafficPolicyInstancesByPolicyCommand };
