// @aws-sdk/client-route-53/dist-es/commands/ListTrafficPolicyInstancesByHostedZoneCommand.js
import { __extends } from "tslib";
import { getIdNormalizerPlugin } from "@aws-sdk/middleware-sdk-route53";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListTrafficPolicyInstancesByHostedZoneRequestFilterSensitiveLog, ListTrafficPolicyInstancesByHostedZoneResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restXmlListTrafficPolicyInstancesByHostedZoneCommand, serializeAws_restXmlListTrafficPolicyInstancesByHostedZoneCommand, } from "../protocols/Aws_restXml";
var ListTrafficPolicyInstancesByHostedZoneCommand = (function (_super) {
    __extends(ListTrafficPolicyInstancesByHostedZoneCommand, _super);
    function ListTrafficPolicyInstancesByHostedZoneCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListTrafficPolicyInstancesByHostedZoneCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getIdNormalizerPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "Route53Client";
        var commandName = "ListTrafficPolicyInstancesByHostedZoneCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListTrafficPolicyInstancesByHostedZoneRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ListTrafficPolicyInstancesByHostedZoneResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListTrafficPolicyInstancesByHostedZoneCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlListTrafficPolicyInstancesByHostedZoneCommand(input, context);
    };
    ListTrafficPolicyInstancesByHostedZoneCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlListTrafficPolicyInstancesByHostedZoneCommand(output, context);
    };
    return ListTrafficPolicyInstancesByHostedZoneCommand;
}($Command));
export { ListTrafficPolicyInstancesByHostedZoneCommand };
