// @aws-sdk/client-route-53/dist-es/commands/ListHostedZonesByVPCCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListHostedZonesByVPCRequestFilterSensitiveLog, ListHostedZonesByVPCResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restXmlListHostedZonesByVPCCommand, serializeAws_restXmlListHostedZonesByVPCCommand, } from "../protocols/Aws_restXml";
var ListHostedZonesByVPCCommand = (function (_super) {
    __extends(ListHostedZonesByVPCCommand, _super);
    function ListHostedZonesByVPCCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListHostedZonesByVPCCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "Route53Client";
        var commandName = "ListHostedZonesByVPCCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListHostedZonesByVPCRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ListHostedZonesByVPCResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListHostedZonesByVPCCommand.prototype.serialize = function (input, context) {
        return serializeAws_restXmlListHostedZonesByVPCCommand(input, context);
    };
    ListHostedZonesByVPCCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restXmlListHostedZonesByVPCCommand(output, context);
    };
    return ListHostedZonesByVPCCommand;
}($Command));
export { ListHostedZonesByVPCCommand };
